uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/microsoftsts/MicrosoftStsPromptHandler.java

```diff
@@ -76,6 +76,20 @@ public void handlePrompt(@NonNull final String username, @NonNull final String p
             loginComponentHandler.handleEmailField(username);
         }
 
+        if (parameters.getPrompt() == PromptParameter.WHEN_REQUIRED ) {
+            if (parameters.isConsentPageExpected()) {
+                final UiResponse consentPageResponse = parameters.getConsentPageResponse();
+                if (consentPageResponse == UiResponse.ACCEPT) {
+                    loginComponentHandler.acceptConsent();
+                } else {
+                    loginComponentHandler.declineConsent();
+                }
+            }
+            if(parameters.isPasswordPageExpected()) {
+                loginComponentHandler.handlePasswordField(password);
+            }
+        }
+
         if (parameters.getPrompt() == PromptParameter.LOGIN || !parameters.isSessionExpected()) {
             loginComponentHandler.handlePasswordField(password);
         }
```

Comment:
Personally don't like this logic being tied into the `if block for WHEN_REQURIED`...in fact, we probably don't need to check for that at all (we are already doing the consent check further below). I think the change that I've suggested here below should do it.

```suggestion
        if (parameters.isPasswordPageExpected() || parameters.getPrompt() == PromptParameter.LOGIN || !parameters.isSessionExpected()) {
            loginComponentHandler.handlePasswordField(password);
        }
```